tests/Compat/FormDecorator/LabelDecoratorTest.php

```diff
@@ -23,4 +27,95 @@ public function testWithoutLabelAttribute(): void
 
         $this->assertSame('&nbsp;', $results->assemble()->render());
     }
+
+    public function testRequiredElement(): void
+    {
+        $formElement = new TextElement('test', [
+            'required' => true,
+            'label' => 'test-label'
+        ]);
+
+        $results = new FormElementDecorationResult();
+        $this->decorator->decorateFormElement($results, $formElement);
+
+        $html = sprintf(
+            <<<'HTML'
+        <label class="form-element-label" for="%s">test-label
+        <span class="required-hint" aria-hidden="true" title="Required"> *</span></label>
+        HTML,
+            $formElement->getAttribute('id')->getValue(),
+        );
+
+        $this->assertHtml($html, $results->assemble());
+        $this->assertSame($formElement->getAttribute('aria-required')->getValue(), 'true');
+    }
+
+    public function testNonRequiredElement(): void
+    {
+        $formElement = new TextElement('test', [
+            'required' => false,
+            'label' => 'test-label'
+        ]);
+
+        $results = new FormElementDecorationResult();
+        $this->decorator->decorateFormElement($results, $formElement);
+
+        $this->assertStringNotContainsString(
+            '<span class="required-hint" aria-hidden="true" title="Required"> *</span>',
+            $results->assemble()->render()
+        );
+        $this->assertTrue(! $formElement->hasAttribute('aria-required')
+            || $formElement->getAttribute('aria-required')->getValue() !== "true");
```

Comment:
Please simplify this :laughing: `assertFalse` and check only for the attribute's existence.